app/src/androidTest/java/com/github/h3lp3rs/h3lp/ChatUiTest.kt

```diff
@@ -32,28 +42,67 @@ class ChatUiTest {
     private lateinit var conversationFrom: Conversation
     private lateinit var conversationTo: Conversation
 
+    private lateinit var db: Database
+
+    private lateinit var foreignUserPublicKey: PublicKey
+    private lateinit var foreignUserPrivateKey: PrivateKey
+
+
     @Before
     fun setup() {
         // Launching the activity with the needed parameters
         val intent = Intent(
-            ApplicationProvider.getApplicationContext(),
+            getApplicationContext(),
             ChatActivity::class.java
         ).apply {
             putExtra(EXTRA_CONVERSATION_ID, CONVERSATION_ID)
             putExtra(EXTRA_USER_ROLE, Messenger.HELPEE)
         }
 
-        ActivityScenario.launch<ChatActivity>(intent)
+        val kpg =
+            KeyPairGenerator.getInstance(
+                KeyProperties.KEY_ALGORITHM_RSA,
+                Conversation.ANDROID_KEY_STORE
+            )
+
+        kpg.initialize(
+            KeyGenParameterSpec.Builder(
+                "key", KeyProperties.PURPOSE_ENCRYPT
+                        or KeyProperties.PURPOSE_DECRYPT
+            )
+                .setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_RSA_OAEP)
+                .setDigests(KeyProperties.DIGEST_SHA1)
```

Comment:
BEWARE!
SHA1 has been cracked and is no longer safe, maybe you should use another hash function? (ex: Keccak or SHA3).